packages/automerge-repo/test/DocHandle.test.ts

```diff
@@ -325,6 +326,52 @@ describe("DocHandle", () => {
     assert.equal(handle.isDeleted(), true)
   })
 
+  it("should clear document when reset", async () => {
+    const handle = setup()
+
+    handle.change(doc => {
+      doc.foo = "bar"
+    })
+    const doc = await handle.doc()
+    assert.equal(doc?.foo, "bar")
+
+    handle.reset()
+    assert(handle.inState([IDLE]))
+    const clearedDoc = await handle.doc([IDLE])
+    assert.notEqual(clearedDoc?.foo, "bar")
+  })
+
+  it("should allow reloading after reset and being idle", async () => {
+    // set docHandle time out after 5 ms
+    const handle = setup({ timeoutDelay: 5 })
+
+    handle.change(doc => {
+      doc.foo = "bar"
+    })
+    const doc = await handle.doc()
+    assert.equal(doc?.foo, "bar")
+
+    handle.reset()
+
+    // remain idle beyond timeoutDelay
+    await pause(10)
```

Comment:
I think we should assert that we are in the IDLE state after this call, otherwise this test passes even if `reset` is a no-op.